org.civicrm.reqangular/src/common/services/format-currency.service.js

```diff
@@ -21,84 +22,96 @@ define([
+     * Prepare regex expression for special characters as following:
+     * for characters '\\' and ']'
      *
-     * @param  {string} decimalSeparator
-     * @return {string}
+     * @param  {String} decimalSeparator
+     * @return {String} regex expression
      */
-    function configureExpression (decimalSeparator) {
-      var expression;
+    function prepareRegexForSeparator (decimalSeparator) {
+      var escape = ['\\', ']'];
 
-      // Handle reserved regex characters that might be used as separators
-      switch (decimalSeparator) {
-        case '\\':
-          expression = '[^0-9\\\\]';
-          break;
-        case ']':
-          expression = '[^0-9\\]]';
-          break;
-        default:
-          expression = '[^0-9' + decimalSeparator + ']';
-      }
+      _.includes(escape, decimalSeparator) && (decimalSeparator = '\\' + decimalSeparator);
 
-      return expression;
+      return '[^0-9' + decimalSeparator + ']';
     }
 
     /**
      * Formats the given amount with thousand separators
      *
-     * @param  {string|integer} amount
-     * @param  {object} separators
-     * @return {string}
+     * @param  {String|Number} amount
+     * @param  {Object} separators
+     * @return {String}
      */
     function addSeparators (amount, separators) {
-      var amountAfterDecimal; // Keep 2 decimal values after decimal
-      var amountBeforeDecimal;
-      var decimalIndex;
-      var expression;
+      return addThousandSeparator(replaceDecimalSeparator(amount, separators.decimal), separators);
+    }
 
-      // Handling special case:
-      // If we are given to format the amount with decimal as "." ,
-      // we need to change "." into defined decimal separator for proper formatting
-      if (+amount && (amount.toString()).indexOf('.') !== -1) {
-        amount = (amount.toString()).replace('.', separators.decimal);
+    /**
+     * Replaces the decimal separator "." by defined decimal separator
+     *
+     * Add given amount is floating point value with decimal separator as "." ,
+     * we need to change "." into defined decimal separator for proper formatting
+     *
+     * @param {String|Number} amount floating point number  (eg. 123213.98 or 123213.98)
+     * @param {String} decimalSeparator defined decimal separator
+     * @returns {String}
+     */
+    function replaceDecimalSeparator (amount, decimalSeparator) {
+      amount = amount.toString();
+
+      if (+amount && amount.indexOf('.') !== -1) {
+        amount = amount.replace('.', decimalSeparator);
       }
 
-      // Regex expression to remove all non numeric values
-      // except defined decimal separator in the string
-      expression = configureExpression(separators.decimal);
+      return amount;
+    }
+
+    /**
+     * Add thousand separators to the given amount.
+     *
+     * @param {String} amount
+     * @return {String} amount separated with correct separators
```

Comment:
In that particular case `@return {String}` would be enough since the description is enough.